### PR TITLE
Add phpDoc to public methods on classes

### PR DESCRIPTION
**Pull request by navitronic**
This PR adds phpDoc to the public methods of most classes and tidies up some internal logic for clarity.

Reasoning: It's nice to know what classes things expect / refer to when using PHP.

### CODE REVIEW

**Comment by vinkla** [comment]
LGTM. Though, I'd argue we should use fullpath for classes in the docblocs.

File: src/Algorithm.php [hunk 1]
```diff
@@ -4,11 +4,11 @@
 
 class Algorithm
 {
-    private function __construct()
-    {
-        // static class.
-    }
-
```
Comment (pda) [inline src/Algorithm.php]: This deletion doesn't look related to this PR.

**Comment by pda** [comment]
This would be easier to evaluate as two pull requests; one for docblock, one for refactoring / design changes.

**Comment by navitronic** [comment]
@pda fair point, I'll move the logic changes into a separate PR.

**Comment by navitronic** [comment]
@pda this should be ready for a proper review now.

File: src/SymfonyRequestMessage.php [hunk 2]
```diff
@@ -2,22 +2,37 @@
 
 namespace HttpSignatures;
 
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\HttpFoundation\Request;
```
Comment (pda) [inline src/SymfonyRequestMessage.php]: Are these here just for the phpDoc `@var` things? Is that normal?

File: src/SymfonyRequestMessage.php [hunk 3]
```diff
@@ -2,22 +2,37 @@
...
     private $request;
+
+    /** @var ParameterBag */
     public $headers;
 
-    public function __construct($request)
+    /**
+     * @param Request $request
+     */
+    public function __construct(Request $request)
```
Comment (pda) [inline src/SymfonyRequestMessage.php]: I don't think this belongs in this PR.

**Comment by pda** [comment]
Looking good aside from my line-notes :+1:

File: src/SymfonyRequestMessage.php (hunk 2 above)
Comment (navitronic) [inline src/SymfonyRequestMessage.php]: It's how I've tended to do it and seen it done: http://www.phpdoc.org/docs/latest/guides/types.html#class-name

Comment (pda) [inline src/SymfonyRequestMessage.php]: Cool, I also ran it through `php-cs-fixer` (I'll push the result to this branch) and the `use` statements didn't vanish, so I guess that's considered normal.

Comment (pda) [inline src/SymfonyRequestMessage.php]: Oh, too late, it's merged.

Comment (navitronic) [inline src/SymfonyRequestMessage.php]: Ah, sorry.

Might be good to get those fixes separately anyhow.
